util/helm/creds.go

```diff
@@ -0,0 +1,257 @@
+package helm
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+
+	argoutils "github.com/argoproj/argo-cd/v2/util"
+	"github.com/argoproj/argo-cd/v2/util/env"
+	"github.com/argoproj/argo-cd/v2/util/workloadidentity"
+)
+
+// In memory cache for storing Azure tokens
+var azureTokenCache *gocache.Cache
+
+func init() {
+	azureTokenCache = gocache.New(gocache.NoExpiration, 0)
+}
+
+// StoreToken stores a token in the cache
+func storeAzureToken(key, token string, expiration time.Duration) {
+	azureTokenCache.Set(key, token, expiration)
+}
+
+type Creds interface {
+	GetUsername() string
+	GetPassword() (string, error)
+	GetCAPath() string
+	GetCertData() []byte
+	GetKeyData() []byte
+	GetInsecureSkipVerify() bool
+}
+
+type HelmCreds struct {
+	Username           string
+	Password           string
+	CAPath             string
+	CertData           []byte
+	KeyData            []byte
+	InsecureSkipVerify bool
+}
+
+func (creds HelmCreds) GetUsername() string {
+	return creds.Username
+}
+
+func (creds HelmCreds) GetPassword() (string, error) {
+	return creds.Password, nil
+}
+
+func (creds HelmCreds) GetCAPath() string {
+	return creds.CAPath
+}
+
+func (creds HelmCreds) GetCertData() []byte {
+	return creds.CertData
+}
+
+func (creds HelmCreds) GetKeyData() []byte {
+	return creds.KeyData
+}
+
+func (creds HelmCreds) GetInsecureSkipVerify() bool {
+	return creds.InsecureSkipVerify
+}
+
+type AzureWorkloadIdentityCreds struct {
+	repoUrl            string
+	CAPath             string
+	CertData           []byte
+	KeyData            []byte
+	InsecureSkipVerify bool
+	tokenProvider      workloadidentity.TokenProvider
+}
+
+func (creds AzureWorkloadIdentityCreds) GetUsername() string {
+	return "00000000-0000-0000-0000-000000000000"
+}
```

Comment:
Why do we need this method at all? Is it possible to refactor with maybe using inheritance so that only relevant methods are available?

Reply:
00000000-0000-0000-0000-000000000000 is a valid user name, to login to helm we need to use 00000000-0000-0000-0000-000000000000 as user name and password is the accessToken from ACR.

This method is kept to allow overload for defining the user name for the workload identity credentials